server/aesdsocket.c

```diff
@@ -105,14 +177,124 @@ void daemonize() {
         }
     }
 
-   
+
 }
 
-int main(int argc, char *argv[]) {
+void *thread_function(void *thread_param) {
+
+    //Get the params
+    thread_data_t *threadData = (thread_data_t *) thread_param;
+    int client_socket = threadData->client_socket;
+    ssize_t bytes_received;
+    char recv_buffer[BUFFER_SIZE];
+
+    //Setup the temp file name
+    char temp_file[256];
+    snprintf(temp_file, sizeof(temp_file), "/var/tmp/tempfile_%d.txt", threadData->thread_num);
```

Comment:
@Silex93 I see that you are using temporary files per thread to avoid the need to dynamically allocate memory.  It seems it would be easier to use dynamic allocation, however.